### PR TITLE
feat(monitors): Support sending monitor check-ins as JSON payload

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Implements https://github.com/getsentry/relay/issues/2334 + https://github.com/getsentry/relay/issues/2333

---

Thank you very much @jan-auer for pairing with me on this!

### CODE REVIEW

File: relay-monitors/src/lib.rs [hunk 1]
```diff
@@ -129,6 +129,7 @@ pub struct CheckIn {
     pub check_in_id: Uuid,
 
     /// Identifier of the monitor for this check-in.
+    #[serde(default)]
     pub monitor_slug: String,
```
Comment (evanpurkhiser) [inline relay-monitors/src/lib.rs:133]: Slug is now no longer required since it is not required when using the `POST` API endpoint.

Theoretically you could already send the empty string as @jan-auer pointed out, so we probably don't need to handle an error condition for this missing

File: tests/integration/test_monitors.py [hunk 2]
```diff
@@ -10,6 +11,49 @@ def generate_check_in(slug):
     }
 
 
+def test_monitor_query_params(mini_sentry, relay):
```
Comment (evanpurkhiser) [inline tests/integration/test_monitors.py]: This isn't actually using a query string, this is ingesting the monitor as an envelope, this can probably be split into a separate PR

**Review by jan-auer — APPROVED** [review]
Good to merge once a changelog is added.

File: tests/integration/test_monitors.py (hunk 2 above)
Comment (evanpurkhiser) [reply tests/integration/test_monitors.py]: #2479
